src/server/pfs/server/driver.go

```diff
@@ -321,100 +320,96 @@ func (d *driver) deleteAllBranchesFromRepos(txnCtx *txncontext.TransactionContex
 	}
 	return nil
 }
-
 func (d *driver) deleteRepo(txnCtx *txncontext.TransactionContext, repo *pfs.Repo, force bool) error {
 	repos := d.repos.ReadWrite(txnCtx.SqlTx)
-
-	// check if 'repo' is already gone. If so, return that error. Otherwise,
-	// proceed with auth check (avoids awkward "access denied" error when calling
-	// "deleteRepo" on a repo that's already gone)
-	var repoInfo pfs.RepoInfo
-	err := repos.Get(repo, &repoInfo)
-	if err != nil {
+	repoInfo := &pfs.RepoInfo{}
+	if err := repos.Get(repo, repoInfo); err != nil {
 		if !col.IsErrNotFound(err) {
 			return errors.Wrapf(err, "error checking whether %q exists", repo)
 		}
 	}
+	return d.deleteRepos(txnCtx, []*pfs.RepoInfo{repoInfo}, force)
+}
 
-	// Check if the caller is authorized to delete this repo
-	if err := d.env.AuthServer.CheckRepoIsAuthorizedInTransaction(txnCtx, repo, auth.Permission_REPO_DELETE); err != nil {
-		return errors.EnsureStack(err)
-	}
-
-	if !force {
-		if _, err := d.env.GetPPSServer().InspectPipelineInTransaction(txnCtx, pps.RepoPipeline(repo)); err == nil {
-			return errors.Errorf("cannot delete a repo associated with a pipeline - delete the pipeline instead")
-		} else if err != nil && !errutil.IsNotFoundError(err) {
+// NOTE: repoInfos should all exist in the system, or else we'll get an awkward ACCESS DENIED error when doing the auth check
```

Comment:
Should this be part of the provenance work?

It seems like the earlier approach of hiding the spurious message was better TBH.

Reply:
this is actually something that I haven't changed just reformatted. I think it's okay to leave this for now